src/librustdoc/clean/mod.rs

```diff
@@ -627,12 +630,38 @@ fn clean_generic_param<'tcx>(
             } else {
                 ThinVec::new()
             };
+
+            // If this ends up being slow, then optimize it by reading the local bounds
+            // (from all predicate origins) and check if a bound on `?Sized` is present.
+            // If there's no `?Sized` bound, then definitely `allow_unsized = false`.
+            let allow_unsized = {
```

Comment:
@fmease would you mind reviewing this bit? I'm not familiar enough with how rustc/rustdoc represent generic bounds to be confident saying this is correct.